configure.ac

```diff
@@ -2250,12 +2254,22 @@ AS_IF([test x"$enable_installing_source_artifacts" = "xno"],
   [install_source_artifacts=true])
 
 AS_IF([test x"$enable_ocamldoc" = "xno"],
-  [ocamldoc=""
+  [AS_IF([test x"$enable_stdlib_manpages" = "xyes"],
+    [AC_MSG_ERROR([--enable-stdlib-manpages requires ocamldoc])])
```

Comment:
I would put this test much higher up, around L532 just after the checks for the dependencies of ocamldoc - the reason is that `./configure --disable-ocamldoc --enable-stdlib-manpages` does an awful lot of work before returning the error relative to `./configure --disable-unix-lib --enable-ocamldoc`